controllers/xrootd/xrootdcluster_controller_test.go

```diff
@@ -0,0 +1,208 @@
+/*
+
+
+Copyright (C) 2020  The XRootD Collaboration
+
+This library is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation; either
+version 2.1 of the License, or (at your option) any later version.
+
+This library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+USA
+*/
+
+package xrootdcontroller
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	catalogv1alpha1 "github.com/xrootd/xrootd-k8s-operator/apis/catalog/v1alpha1"
+	xrootdv1alpha1 "github.com/xrootd/xrootd-k8s-operator/apis/xrootd/v1alpha1"
+	"github.com/xrootd/xrootd-k8s-operator/tests/integration/framework"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func applyCrsAndFetchCrAfterDelay(
+	versionCr *catalogv1alpha1.XrootdVersion,
+	clusterCr *xrootdv1alpha1.XrootdCluster,
+	clusterKey types.NamespacedName,
+	delay time.Duration,
+) *xrootdv1alpha1.XrootdCluster {
+	By("creating xrootdversion successfully")
+	Expect(testFramework.Client.Create(context.TODO(), versionCr)).Should(Succeed())
+	By("creating xrootdcluster successfully")
+	Expect(testFramework.Client.Create(context.TODO(), clusterCr)).Should(Succeed())
+	time.Sleep(delay)
+
+	By("fetching updated xrootdcluster CR")
+	fetched := &xrootdv1alpha1.XrootdCluster{}
+	Expect(testFramework.Client.Get(context.TODO(), clusterKey, fetched)).Should(Succeed())
+	return fetched
+}
+
+var _ = Describe("XrootdCluster Controller", func() {
+	const waitCr = time.Second * 5
+
+	var (
+		versionSpec     catalogv1alpha1.XrootdVersionSpec
+		versionToCreate *catalogv1alpha1.XrootdVersion
+		clusterSpec     xrootdv1alpha1.XrootdClusterSpec
+		clusterToCreate *xrootdv1alpha1.XrootdCluster
+		versionKey      types.NamespacedName
+		clusterKey      types.NamespacedName
+	)
+
+	// sets up test framework
+	testFramework.InitOnRunningSuite()
+
+	BeforeEach(func() {
+		clusterKey = types.NamespacedName{
+			Namespace: testFramework.GetNamespace(),
```

Comment:
The test framework also creates an ephemeral network for each test spec so it is guaranteed no resources would remain undeleted